lib/kubeclient/common.rb

```diff
@@ -165,9 +175,10 @@ def watch_entities(entity_type, options = {})
       path += "/#{options[:name]}" if options[:name]
       uri = @api_endpoint.merge("#{@api_endpoint.path}/#{@api_version}/#{path}")
 
-      params = options.slice(:label_selector, :field_selector, :resource_version)
+      params = {}
+      WATCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
       if params.any?
-        uri.query = URI.encode_www_form(params.map { |k, v| [k.to_s.camelize(:lower), v] })
+        uri.query = URI.encode_www_form(params.map { |k, v| [k, v] })
```

Comment:
Actually, now that I see it, you can pass params directly into the encode_www_form without translation into an intermediate Array:

```ruby
[1] pry(main)> params = {:a => 1, :b => 2}
=> {:a=>1, :b=>2}
[2] pry(main)> URI.encode_www_form(params.to_a)
=> "a=1&b=2"
[3] pry(main)> URI.encode_www_form(params)
=> "a=1&b=2"
```